ts/api/backend.ts

```diff
@@ -0,0 +1,172 @@
+const FullProfile = t.intersection([ExtendedProfile, LimitedProfile]);
+
+//
+// Define the types of the requests
+//
+
+type GetServiceT = IGetApiRequestType<
+  {
+    id: string;
+  },
+  "Authorization",
+  never,
+  BasicResponseType<ServicePublic>
+>;
+
+type GetMessagesT = IGetApiRequestType<
+  {
+    cursor?: number;
+  },
+  "Authorization",
+  "cursor",
+  BasicResponseType<Messages>
+>;
+
+type GetMessageT = IGetApiRequestType<
+  {
+    id: string;
+  },
+  "Authorization",
+  never,
+  BasicResponseType<MessageWithContent>
+>;
+
+type GetProfileT = IGetApiRequestType<
+  {},
+  "Authorization",
+  never,
+  BasicResponseType<ProfileWithEmail | ProfileWithoutEmail>
+>;
+
+type CreateOrUpdateProfileT = IPostApiRequestType<
+  {
+    newProfile: ExtendedProfile;
+  },
+  "Authorization",
+  never,
+  BasicResponseType<LimitedProfile | ExtendedProfile>
+>;
+
+type CreateOrUpdateInstallationT = IPutApiRequestType<
+  {
+    id: string;
+    installation: Installation;
+  },
+  "Authorization",
+  never,
+  BasicResponseType<Installation>
+>;
+
+//
+// Create client
+//
+
+export function BackendClient(baseUrl: string, token: string) {
+  const options = {
+    baseUrl
+  };
+
+  const tokenHeaderProducer = AuthorizationBearerHeaderProducer(token);
+
+  const getServiceT: GetServiceT = {
+    method: "get",
+    url: params => `/api/v1/services/${params.id}`,
+    query: _ => ({}),
+    headers: tokenHeaderProducer,
+    response_decoder: basicResponseDecoder(ServicePublic)
+  };
+
+  const getMessagesT: GetMessagesT = {
+    method: "get",
+    url: () => `/api/v1/messages`,
+    query: params => ({
+      cursor: params.cursor ? `${params.cursor}` : ""
+    }),
+    headers: tokenHeaderProducer,
+    response_decoder: basicResponseDecoder(Messages)
+  };
+
+  const getMessageT: GetMessageT = {
+    method: "get",
+    url: params => `/api/v1/messages/${params.id}`,
+    query: _ => ({}),
+    headers: tokenHeaderProducer,
+    response_decoder: basicResponseDecoder(MessageWithContent)
+  };
+
+  const getProfileT: GetProfileT = {
+    method: "get",
+    url: () => "/api/v1/profile",
+    query: _ => ({}),
+    headers: tokenHeaderProducer,
+    response_decoder: basicResponseDecoder(ProfileWithOrWithoutEmail)
+  };
+
+  const createOrUpdateProfileT: CreateOrUpdateProfileT = {
+    method: "post",
+    url: () => "/api/v1/profile",
+    headers: composeHeaderProducers(tokenHeaderProducer, ApiHeaderJson),
+    query: _ => ({}),
+    body: p => JSON.stringify(p.newProfile),
+    response_decoder: basicResponseDecoder(FullProfile)
+  };
+
+  const createOrUpdateInstallationT: CreateOrUpdateInstallationT = {
+    method: "put",
+    url: params => `/api/v1/installations/${params.id}`,
+    headers: composeHeaderProducers(tokenHeaderProducer, ApiHeaderJson),
+    query: _ => ({}),
+    body: p => JSON.stringify(p.installation),
+    response_decoder: basicResponseDecoder(Installation)
```

Comment:
The endpoint returns 200 with no content

Reply:
yes there was a bug actually, it's fixed in https://github.com/teamdigitale/italia-ts-commons/pull/22 - once we merge it I will change the code to use `italia-ts-commons`

Reply:
it should be fixed now

Reply:
sorry I understand now what you meant - shouldn't a `CreateOrUpdateInstallation` operation return an `Installation` ? @lussoluca